src/store/index.js

```diff
@@ -34,17 +39,28 @@ export default new Vuex.Store({
       } catch (error) {
         console.log(error.message);
       }
+    },
+
+    //get one restaurant by ID
+    async fetchRestaurantById({ commit }, ID) {
+      try {
+        const response = await axios.get("/restaurant", {
+          params: {
+            res_id: ID
+          }
+        });
+
+        commit("setCurrentRestaurant", response.data);
+      } catch (error) {
+        console.log(error.message);
+      }
```

Comment:
It does fix the redirection problem, via interceptor. Two problems here though:

1. I already fetched quite a big list of restaurants but not I need to re-fetch a single one each time I want to see the details? It will drain my Zomato key pretty quickly (many API keys have limits on how many queries you may perform in a unit of time), don't you think?

2. I went to "/about/16536386" and successfully viewed the details of Cafe La Ruina, a top-rated cafe in Poznań. But your service was supposed to inform me about food places in Wrocław, not Poznań, wasn't it?

Reply:
I added cache to not drain the key. 
The problem is I got stuck with showing only Wroclaw's food places.
When I use 'restaurant' the only parameter you can use is ID of the restaurant.  
I have no clue what to do.

Reply:
Hint: look at the [Zomato API swagger](https://developers.zomato.com/documentation).